ogc/query-with-cql-filters/src/main/java/com/esri/samples/query_with_cql_filters/QueryWithCqlFiltersController.java

```diff
@@ -0,0 +1,316 @@
+    cqlQueryParameters.setWhereClause(comboBox.getSelectionModel().getSelectedItem());
+    // set the max features to the number entered in the text field
+    cqlQueryParameters.setMaxFeatures(Integer.parseInt(textField.getText()));
+
+    // if the time extent checkbox is selected, retrieve the date selected from the date picker and set it to the
+    // query parameters time extent
+    if (timeExtentCheckBox.isSelected()) {
+      // get the value selected from the date picker
+      LocalDate startDate = startDatePicker.getValue();
+      LocalDate endDate = endDatePicker.getValue();
+
+      // convert the value into a calendar
+      Calendar start = new Calendar.Builder().setDate(startDate.getYear(), startDate.getMonthValue(), startDate.getDayOfMonth()).build();
+      Calendar end = new Calendar.Builder().setDate(endDate.getYear(), endDate.getMonthValue(), endDate.getDayOfMonth()).build();
+
+      // set the query parameters time extent
+      cqlQueryParameters.setTimeExtent(new TimeExtent(start, end));
+    }
+
+    // populate and load the table with the query parameters
+    // set the clearCache parameter to true to clear existing table entries and set the outfields parameter to null requests all fields
+    ListenableFuture<FeatureQueryResult> result = ogcFeatureCollectionTable.populateFromServiceAsync(cqlQueryParameters, true, null);
+    result.addDoneListener(() -> {
+
+      // display number of features returned
+      setFeatureNumberLabelText();
+
+      try {
+        // create a new list to store returned geometries in
+        List<Geometry> featureGeometryList = new ArrayList<>();
+
+        // iterate through each result to get its geometry and add it to the geometry list
+        result.get().iterator().forEachRemaining(feature -> {
+          featureGeometryList.add(feature.getGeometry());
+          feature.getGeometry();
+        });
+
+        if (!featureGeometryList.isEmpty()) {
+          // zoom to the total extent of the geometries returned by the query
+          var totalExtent = GeometryEngine.combineExtents(featureGeometryList);
+          mapView.setViewpointGeometryAsync(totalExtent, 20);
+        }
+
+      } catch (InterruptedException | ExecutionException e) {
+        e.printStackTrace();
+      }
+
+      revertToInitialQueryButton.setDisable(false);
+
+    });
+
+  }
+
+  /**
+   * Handles user interaction with the date picker to ensure the end date time is not earlier than the start date.
+   */
+  @FXML
+  private void handleDatePickerInteraction() {
+
+    if (startDatePicker.getValue().toEpochDay() > endDatePicker.getValue().toEpochDay()) {
+      endDatePicker.setValue(startDatePicker.getValue());
+    }
+
+  }
+
+  /**
+   * Handle checkbox interaction to control if the date picker is enabled or not.
+   */
+  @FXML
+  private void handleCheckBoxInteraction() {
+
+    startDatePicker.setDisable(!timeExtentCheckBox.isSelected());
```

Comment:
You might be able to do this by binding when you set up the UI.

Reply:
were you thinking something like this? if so it works, I'm not sure if it's as readable as the above as to what it's doing. keen to make proper use of javafx patterns though. what do you think? 

```java

  private void handleCheckBoxInteraction() {

    startDatePicker.disableProperty().bind(timeExtentCheckBox.selectedProperty().not());
    endDatePicker.disableProperty().bind(timeExtentCheckBox.selectedProperty().not());

  }
```